inference/core/workflows/core_steps/common/utils.py

```diff
@@ -109,40 +146,52 @@ def anchor_detections_in_parent_coordinates(
         image[ORIGIN_COORDINATES_KEY][LEFT_TOP_X_KEY],
         image[ORIGIN_COORDINATES_KEY][LEFT_TOP_Y_KEY],
     )
-    for detection in prediction[f"{detections_key}{PARENT_COORDINATES_SUFFIX}"]:
-        detection["x"] += shift_x
-        detection["y"] += shift_y
-        for point in detection.get("points", []):
-            point["x"] += shift_x
-            point["y"] += shift_y
-        for point in detection.get("keypoints", []):
-            point["x"] += shift_x
-            point["y"] += shift_y
+    anchored_detections: sv.Detections = prediction[
+        f"{detections_key}{PARENT_COORDINATES_SUFFIX}"
+    ]
+    anchored_detections.xyxy += [shift_x, shift_y, shift_x, shift_y]
+    # TODO: assumed type
+    if keypoints_key in anchored_detections.data:
+        anchored_detections[keypoints_key] += [shift_x, shift_y]
+    if anchored_detections.mask:
+        origin_width = image[ORIGIN_COORDINATES_KEY][ORIGIN_SIZE_KEY][WIDTH_KEY]
+        origin_height = image[ORIGIN_COORDINATES_KEY][ORIGIN_SIZE_KEY][HEIGHT_KEY]
+        origin_mask_base = np.full((origin_height, origin_width), False)
+        anchored_detections.mask = [
+            origin_mask_base.copy() for _ in anchored_detections
+        ]
+        for anchored_mask, original_mask in zip(
+            anchored_detections.mask, prediction[detections_key].mask
+        ):
+            mask_h, mask_w = original_mask.shape
+            # TODO: instead of shifting mask we could store contours in data instead of storing mask (even if calculated)
+            #       it would be faster to shift contours but at expense of having to remember to generate mask from contour when it's needed
+            anchored_mask[shift_x : shift_x + mask_w, shift_y : shift_y + mask_h] = (
+                original_mask
+            )
     prediction[f"{image_metadata_key}{PARENT_COORDINATES_SUFFIX}"] = image[
         ORIGIN_COORDINATES_KEY
     ][ORIGIN_SIZE_KEY]
     return prediction
 
 
 def filter_out_unwanted_classes_from_predictions_detections(
-    predictions: List[Dict[str, Any]],
+    predictions: List[Dict[str, Union[sv.Detections, Any]]],
     classes_to_accept: Optional[List[str]],
     detections_key: str = "predictions",
-    class_name_key: str = "class",
-) -> List[Dict[str, Any]]:
+    class_name_key: str = "class_name",
+) -> List[Dict[str, Union[sv.Detections, Any]]]:
     if classes_to_accept is None:
```

Comment:
that's my mistake, could u make that `if not classes_to_accept` to make empty list case handled better